apps/server/test/modules/files-storage/file-storage-with-s3-server.e2e-spec.ts

```diff
@@ -0,0 +1,294 @@
+import { FileRecordResponse } from '@src/modules/files-storage/controller/dto';
+import { JwtAuthGuard } from '@src/modules/authentication/guard/jwt-auth.guard';
+import { ICurrentUser } from '@shared/domain';
+import { userFactory, roleFactory, cleanupCollections, mapUserToCurrentUser, fileRecordFactory } from '@shared/testing';
+import { ApiValidationError } from '@shared/common';
+import S3rver from 's3rver';
+
+class API {
+	app: INestApplication;
+
+	constructor(app: INestApplication) {
+		this.app = app;
+	}
+
+	async postUploadFile(routeName: string, query?: string | Record<string, unknown>) {
+		const response = await request(this.app.getHttpServer())
+			.post(`/files-storage${routeName}`)
+			.attach('file', Buffer.from('abcd'), 'test.txt')
+			.set('connection', 'keep-alive')
+			.set('content-type', 'multipart/form-data; boundary=----WebKitFormBoundaryiBMuOC0HyZ3YnA20')
+			.query(query || {});
+
+		return {
+			result: response.body as FileRecordResponse,
+			error: response.body as ApiValidationError,
+			status: response.status,
+		};
+	}
+
+	async getDownloadFile(routeName: string, query?: string | Record<string, unknown>) {
+		const response = await request(this.app.getHttpServer())
+			.get(`/files-storage${routeName}`)
+			.query(query || {});
+
+		return {
+			result: response.body as FileRecordResponse,
+			error: response.body as ApiValidationError,
+			status: response.status,
+		};
+	}
+}
+
+describe('file-storage controller (e2e)', () => {
+	let app: INestApplication;
+	let orm: MikroORM;
+	let em: EntityManager;
+	let currentUser: ICurrentUser;
+	let api: API;
+	let s3instance: S3rver;
+
+	beforeAll(async () => {
+		s3instance = new S3rver({
+			directory: '/tmp/s3rver_test_directory',
+			resetOnClose: true,
+		});
+		await s3instance.run();
+		const module: TestingModule = await Test.createTestingModule({
+			imports: [FilesStorageTestModule],
+		})
+			.overrideGuard(JwtAuthGuard)
+			.useValue({
+				canActivate(context: ExecutionContext) {
+					const req: Request = context.switchToHttp().getRequest();
+					req.user = currentUser;
+					return true;
+				},
+			})
+			.compile();
+
+		app = module.createNestApplication();
+		await app.init();
+		orm = app.get(MikroORM);
+		em = module.get(EntityManager);
+		api = new API(app);
+	});
+
+	afterAll(async () => {
+		await orm.close();
+		await app.close();
+		await s3instance.close();
+	});
+
+	beforeEach(async () => {
+		await cleanupCollections(em);
+	});
+
+	const setup = () => {
+		const roles = roleFactory.buildList(1, { permissions: [] });
+		const user = userFactory.build({ roles });
+
+		return user;
+	};
+	describe('upload action', () => {
+		describe('with bad request data', () => {
+			it('should return status 400 for invalid schoolId', async () => {
+				const user = setup();
+				const validId = new ObjectId().toHexString();
+
+				await em.persistAndFlush([user]);
+				em.clear();
+
+				currentUser = mapUserToCurrentUser(user);
```

Comment:
The user setup looks like the same in all test cases. Doesn't it make sense to move that to a `beforEach` hook?